docs/book/deploying-zenml/zenml-cloud/cloud-system-architecture.md

```diff
@@ -0,0 +1,79 @@
+
+## Scenario 2a: Hybrid SaaS
+
+<div data-full-width="true">
+<figure><img src="../../.gitbook/assets/cloud_architecture_scenario_2a.png" alt=""><figcaption><p>Scenario 2a: Hybrid SaaS deployment</p></figcaption></figure>
+</div>
+
+This scenario builds on top of Scenario 1, with the exception being that the "Data Artifacts" move over the customer cloud. This can be set up quite easily by setting up an [artifact store](../../stacks-and-components/component-guide/artifact-stores/) with your MLOps stack.
+
+In this scenario, the ZenML Cloud only stores metadata while the customer cloud keeps potentially sensitive customer data. However, if the customer wants access to special ZenML Cloud features like viewing artifacts on the dashboard, triggering pipeline runs, etc then they have to also provide [connectors](../../stacks-and-components/auth-management/service-connectors-guide.md) which are ultimately stored as secrets on the ZenML side. Therefore, while this scenario keeps the data on customer side, it still gives access to ZenML Cloud to trigger actions via an ingress connection over the internet into the customer cloud.
+
+Note, that ZenML encrypts all customer secrets at transit and at rest.
+
+## Scenario 2b: Hybrid SaaS + Private Secret Store
+
+<div data-full-width="true">
+<figure><img src="../../.gitbook/assets/cloud_architecture_scenario_2b.png" alt=""><figcaption><p>Scenario 2a: Hybrid SaaS deployment + Private Secret Store</p></figcaption></figure>
+</div>
+
+This is almost the same as Scenario 2a, however in this the customer brings their own secret store to keep sensitive information on customer side. This allows users to use the [secrets API](../../user-guide/advanced-guide/secret-management/) and ensure their secrets do not leave their environment.
+
+The ZenML secret store still exists, and this is again to enable customers to create [connectors](../../stacks-and-components/auth-management/service-connectors-guide.md) to allow ZenML Cloud to triggers actions on their environment.
+
+## Scenario 3: Agent Architecture
+
+<div data-full-width="true">
+<figure><img src="../../.gitbook/assets/cloud_architecture_scenario_3.png" alt=""><figcaption><p>Scenario 3: ZenML Agent deployment</p></figcaption></figure>
+</div>
+
+This scenario adds a new architectural component into the mix, called the ZenML Agent, which facilitates communication between the two clouds. The customer is responsible for deploying and maintaining the ZenML Agent in their environment.
+
+The advantage of this deployment is that the ZenML Cloud does not need access to any sensitive secrets or need create any [connectors](../../stacks-and-components/auth-management/service-connectors-guide.md) to trigger actions customer side. Secrets and data remain on the customer environment, and only one secure ingress connection is exposed.
```

Comment:
Do we disable viewing data artifacts in the dashboard, to keep the data on the customer cloud? Or perhaps we move the Dashboard to the customer cloud?

Reply:
In the agent architecture, the secrets can be exchanged as short-lived tokens to view the dashboard artifacts I think

Reply:
The agent acts as an intermediate step for all operations and information that needs to be exchanged between ZenML cloud and other customer stack components, like the artifact store. This means that all features like visualizing data artifacts in the dashboard and triggering pipelines from the dashboard are fully available, but only the ZenML Agent has access to customer secrets and accesses the customer's infrastructure services.

More concretely:
1. the ZenML Cloud API asks the agent to fetch an artifact for visualization
2. the agent connects to the S3 bucket directly with the credentials configured by the customer and stored in the customer secret store or via a service connector. Credentials never have to leave the agent.
3. the agent fetches the artifact and sends it back to the Cloud API
4. the Cloud API sends the visualization to the dashboard

Reply:
ill put this in the docs directly